### PR TITLE
Add PIT enabled argumento to Cluster Resource and Data Source

### LINKED ISSUE #104: Add pitEnabled feature of mongodbatlas_cluster resource
I’m trying to automate the deployment of our MongoDB Atlas estate. MongoDB Atlas has an API as part of the cluster configuration which allows to set point in time restore **pitEnabled**

https://docs.atlas.mongodb.com/reference/api/clusters-create-one/

Looking in the documentation for the terraform resource mongodbatlas_cluster it doesn’t have this option

https://www.terraform.io/docs/providers/mongodbatlas/r/cluster.html

Is there a plan to implement this?

### PR DESCRIPTION
**Pull request by marinsalinas**
Closes #104 

This branch is based on the dbusers branch, once that branch is merged I'll rebase from master and then merge it.

### CODE REVIEW

**Review by PacoDw — APPROVED** [review]
LGTM! Thank you so much :)

File: website/docs/r/cluster.html.markdown [hunk 1]
```diff
@@ -235,6 +235,7 @@ Note free tier (M0) creation is not supported by the Atlas API and hence not sup
 * `encryption_at_rest_provider` - (Optional) Set the Encryption at Rest parameter.  Possible values are AWS, GCP, AZURE or NONE.  Requires M10 or greater and for backup_enabled to be false or omitted.
 * `mongo_db_major_version` - (Optional) Version of the cluster to deploy. Atlas supports the following MongoDB versions for M10+ clusters: `3.6`, `4.0`, or `4.2`. You must set this value to `4.2` if `provider_instance_size_name` is either M2 or M5.
 * `num_shards` - (Optional) Selects whether the cluster is a replica set or a sharded cluster. If you use the replicationSpecs parameter, you must set num_shards.
+* `pit_enabled` - (Optional) - Flag that indicates if the cluster uses Point-in-Time backups. If set to true, providerBackupEnabled must also be set to true.
```
Comment (themantissa) [inline website/docs/r/cluster.html.markdown]: One minor change, we should use the Terraform variable so change to:
* `pit_enabled` - (Optional) - Flag that indicates if the cluster uses Point-in-Time backups. If set to true, provider_backup_enabled must also be set to true.

**Review by themantissa — CHANGES_REQUESTED** [review]
Nice work, just one minor doc change and we are good.

**Review by themantissa — APPROVED** [review]
Nicely done, LGTM

**Pull request merged**